src/components/invest/index.js

```diff
@@ -116,26 +123,55 @@ export class Invest extends React.Component {
           initializeAccumulativeData()
             .then(() => getCrowdsaleData(crowdsaleContract))
             .then(() => getAccumulativeCrowdsaleData())
+            .then(() => getCrowdsaleTargetDates())
+            .then(() => this.setTimers())
+            .catch(err => console.log(err))
             .then(() => this.setState({ loading: false }))
-            .catch(err => {
-              this.setState({ loading: false })
-              console.log(err)
-            })
-
-          getCrowdsaleTargetDates(this, () => {
-            if (crowdsalePageStore.endDate) {
-              this.setState({
-                seconds: (crowdsalePageStore.endDate - new Date().getTime()) / 1000
-              })
-            }
-          })
         })
       })
     })
   }
 
+  setTimers = () => {
+    const { crowdsalePageStore } = this.props
+    let nextTick = 0
+    let millisecondsToNextTic = 0
+    let timeInterval
+
+    if (crowdsalePageStore.ticks.length) {
+      nextTick = crowdsalePageStore.extractNextTick()
+      millisecondsToNextTic = nextTick.time - Date.now()
+
+      setTimeout(() => {
+        this.setState({ displaySeconds: true })
+      }, millisecondsToNextTic - 5*6e4)
+
+      timeInterval = setInterval(() => {
```

Comment:
Should we clear this in the `componentWillUnmount` hook?

Reply:
Done. I also did cleared the interval when the countdown finishes